Makefile

```diff
@@ -33,6 +33,10 @@ container-targets: container-image container-gen-content container-render contai
 help: ## Show this help text.
 	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}'
 
+dependencies:
+	npm ci
+	cd themes/docsy/ && npm i
```

Comment:
Does this need adapting for `make container-server`?

Reply:
Yeah, the `Makefile` needs to be updated. The `dependencies` target itself is ok but needs to be added as dependencies on `docker-server`, `container-server` and `production-build`. I have not added them as I wanted to know the sentiment regarding overhauling the `Makefile`. There is a lot of repetition (like git submodule update --init --recursive --depth 1) that can be moved to its own target and used as a dependency. Let me know if its ok to do that and I will make the changes to this PR.

Reply:
Where possible, split out refactorings like that into their own (separate, smaller) PR.

Reply:
Which is why I haven't made any changes to the Makefile. The `dependencies` target though is necessary for the latest Docsy and it was needed to get the `preview-build` target to work and will be needed for the other targets. Essentially, that command creates 2 folders inside `themes`, `FortAwesome` and `twbs`.

Reply:
Does this PR break local container previewing? If so, let's update the README and call out the issue in the PR description.

Reply:
By local container previewing, do you mean using `make container-server`? It doesn't work in `master` itself. It seems there is a permission issue on the `/tmp` folder. The following error is thrown:
> ...
rsync: [receiver] chown "/tmp/src/themes/docsy/userguide/static/images/.version-banner.png.nPONje" failed: Operation not permitted (1)
rsync: [receiver] chown "/tmp/src/themes/docsy/userguide/static/images/.yes.png.iMPnje" failed: Operation not permitted (1)
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1338) [sender=3.3.0]
make: *** [Makefile:80: container-server] Error 23

This issue came up in WSL and I tried a fresh setup in codespaces and the issue is the same. It seems like the issue dooesn't happen if I remove this line https://github.com/kubernetes/contributor-site/blob/f42204c6a4d640c766900b4c98c931a9d259dfb9/Makefile#L83

But running the docker commands manually works as expected and so local container previewing still works.

Reply:
@sftim From your [comment](https://github.com/kubernetes/contributor-site/pull/531?new_mergebox=true#issuecomment-2558507697), I dunno how this convo got resolved but yeah, `container-server` target in the makefile hasn't been updated. In fact, as mentioned in the convo, all the targets needs to be updated to handle the npm dependency. See [this message](https://github.com/kubernetes/contributor-site/pull/531?new_mergebox=true#discussion_r1779454502) to know my primary concerns.

There are multiple ways this can be approached.

1. I can make the changes in the `Makefile` including adding the dependencies and can keep the refactoring for later.
2. I can add the dependencies and do the refactoring.
3. I can do the refactoring in a separate PR to be merged first and fix the dependencies here so that you can get it working.

For now, for you to test things, I will push a commit adding the dependant target to `container-server`.

Reply:
Also, the rsync permission issue is still there in my system. I have pushed a commit that should get `container-server` working. If it doesn't try removing lines 87 and 88 (--cap-drop=ALL and --cap-drop=AUDIT_WRITE). Because I can't get things to work without removing them.

I don't know the real issue behind this as I assume it used to work.

Reply:
We only support container previewing on Linux and macOS host OSs; if you're using another environment, the support may not have been added yet.

Reply:
Hmm...this permission issue occurs on GitHub Codespaces too. But this PR _as of now_ has a new commit that updates the `container-server` target. Can you check if that works for your local host?

I will split this PR into 2 tomorrow.